clash-ghc/src-bin-8.10/Clash/Main.hs

```diff
@@ -116,8 +116,8 @@ import           Clash.GHC.Util (handleClashException)
 -----------------------------------------------------------------------------
 -- GHC's command-line interface
 
-defaultMain :: [String] -> IO ()
-defaultMain = flip withArgs $ do
+defaultMain :: Ghc () -> [String] -> IO ()
+defaultMain startAction = flip withArgs $ do
```

Comment:
Perhaps add `mainWithOpts`? This change would make it more difficult to keep the starter project working across Clash versions.

Reply:
But more importantly, I feel like calling `defaultMain` twice should simply be safe. Is there a way to make this happen?

Reply:
To your first question: sure.
To your second question: No, not without significant effort in removing shared state in GHC.